### ISSUE TITLE
asyngular CL deploy on two clusters (dev and prod)

### ISSUE THREAD

**Issue by hervejegou** [SocketCluster/asyngular-server#3]
We have a 'dev' cluster and are running asyngular server on it, using asyngular CL and it is working great. we did a ```asyngular deploy``` initially (which configured everything locally and in docker and loaded the tls credentials in secret ?)  and we are now doing ```asyngular deploy-update``` on a regular basis. So far it is running great and the performance is really good and fast.

We need to have both a 'production' server/cluster and a 'dev' server. I have added a second cluster 'production' and now I want to deploy the same app to the production cluster (just changed some settings variable). I have setup the second cluster in kubectl and I can switch context between 'dev' and 'production' via kubectl

The question is:
1. do I need to do a first ```asyngular deploy``` for production cluster ? is that going to mess up my 'dev' cluster setup once I switch back to 'dev' context and try to do ```asyngular deploy-update``` for my next 'dev' update ?

2. or can I do ```asyngular deploy-update``` to deploy to 'production' cluster and load manually the agc-tls-credentials to my 'production' cluster ? does 'deploy-update' work the same as 'deploy' except that 'deploy' loads the credentials (and setup docker and local config) ? Since my docker and local configuration is completed, it should work ?

if I can do ```asyngular deploy-update``` to the 'production' cluster, then I can use the same repo and same docker and local config and load it once on 'dev' cluster with 'deploy-update' then switch context to 'production' and do 'deploy-update' again and load to 'production' cluster. the only drawback is I generate new docker images each time but that is not a problem.

Or should I instead load all the yaml files in kubernetes folder one by one via kubectl to 'production' cluster ? then do 'deploy-update' ?

**Comment by jondubois** [comment]
@hervejegou 

1. `asyngular deploy` should always be used if the cluster is new or if there are currently no services and no deployments active on the cluster. So if you previously removed all deployments and services from the cluster (e.g. `asyngular undeploy` or you deleted them using your GKE control panel), then you should recreate your cluster with a fresh `asyngular deploy` command. `asyngular deploy-update` is only  for cases when there are existing deployments and AGC services already running and you want to replace them with new versions with minimal disruption to the existing service.

2. You can use `asyngular add-secret` at any point of your cluster lifecycle but I can't remember whether or not other services will pick up the secret automatically at runtime or if you need to undeploy and redeploy all relevant services. This is more related to K8s than Asyngular.

Basically use `asyngular deploy` if the cluster doesn't have existing AGC services running and use `asyngular deploy-update` if you have old versions of the services running and want to update them without disruption. If you use regular deploy and there are some services still running, then it will just skip those (and you might get an error in your console) but it generally shouldn't cause any major problems. You can always tear down all services and start again if you run into any unexpected issues.

**Comment by hervejegou** [comment]
@jondubois  thanks for the details we will try it shortly. server is running great on GCP and never crashes !
